src/rust/src/x509/verify.rs

```diff
@@ -169,6 +200,7 @@ enum SubjectOwner {
     // so, which was only stabilized with 3.10.
     DNSName(String),
     IPAddress(pyo3::Py<pyo3::types::PyBytes>),
+    None,
```

Comment:
Flagging: This is a bit of a hack; it's the simplest way I could think of to get the "no subject provided" state into the `SubjectOwner` type, which is then required to erase the lifetime on `Policy<'_>` (via `OwnedPolicy`).